dformat: init at 0.15.2

ipsavitsky opened the pull request:
This change introduces `dfmt` - a popular formatter for the D language.

> [!WARNING]
> Because the name of this package (`dfmt`) clashes with another package in nixpkgs, the derivation name is set as `dformat`, which does not appear as an official name of this package anywhere. If there is a better way to resolve this naming conflict - please let me know 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ipsavitsky committed:
dformat: init at 0.15.2

ethancedwards8 commented on pkgs/by-name/df/dformat/package.nix:
```diff
@@ -0,0 +1,33 @@
+  src = fetchFromGitHub {
+    owner = "dlang-community";
+    repo = "dfmt";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

ipsavitsky replied:
Nice, fixed

ethancedwards8 commented on pkgs/by-name/df/dformat/package.nix:
```diff
@@ -0,0 +1,33 @@
+  meta = {
+    description = "Formatter for D source code";
+    homepage = "https://github.com/dlang-community/dfmt";
+    license = lib.licenses.boost;
```
add yourself as a maintainer or find a willing adopter, please.

ipsavitsky replied:
Done, added myself

ethancedwards8 commented on pkgs/by-name/df/dformat/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  meta = {
+    description = "Formatter for D source code";
+    homepage = "https://github.com/dlang-community/dfmt";
```
```suggestion
    homepage = "https://github.com/dlang-community/dfmt";
    changelog = "https://github.com/dlang-community/dfmt/releases/tag/v${version}";
```

ipsavitsky replied:
Fixed

ethancedwards8 reviewed (changes requested):
also consider adding nix-update-script.

no comment on the name conflict thing. it makes sense to me but i'm not an authority on such matters.

overall LGTM. Thank you for contributing!

awwpotato commented on pkgs/by-name/df/dformat/package.nix:
```diff
@@ -0,0 +1,36 @@
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
using with lib is generally frowned upon

ipsavitsky replied:
Fixed

awwpotato commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401429`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dformat</li>
  </ul>
</details>

0xda157 reviewed (commented):
you might consider fixing the output of `--version` and adding a versionCheckHook

ipsavitsky commented:
@awwpotato I have fixed the `--version` flag and added a `versionCheckHook`. Both are passing. 

When building w/ dub dfmt expects a `bin/dubhash.txt`, which is generated using `git` using git history. `git` is called in `dubhash.d` from dubs `preBuild` phase.

Since there is no git history in the source derivation, I have opted to supply `bin/dubhash.txt` via nix and disabling the prebuild phase in dub by removing it from `dub.json`

awwpotato commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401429`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dformat</li>
  </ul>
</details>
